src/main/java/core/basesyntax/service/impl/ShopFileReaderImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ShopFileReader;
+import java.io.BufferedReader;
+import java.io.IOException;
+
+public class ShopFileReaderImpl implements ShopFileReader {
+    @Override
+    public String read(String path) {
+        StringBuilder builder = new StringBuilder();
+        String line = "";
+        try (java.io.FileReader fr = new java.io.FileReader(path)) {
+            BufferedReader br = new BufferedReader(fr);
```

Comment:
Use try-with-resources to properly close the BufferedReader. Example: 'try (BufferedReader br = new BufferedReader(new FileReader(path)))'.